simulation/traffic_simulator/test/src/utils/test_pose.cpp

```diff
@@ -0,0 +1,670 @@
+  EXPECT_LANELET_POSE_NEAR(
+    static_cast<traffic_simulator::LaneletPose>(canonicalized_pose.value()),
+    static_cast<traffic_simulator::LaneletPose>(lanelet_pose), 0.01);
+}
+
+/**
+ * @note Test calculation correctness with a non trivial example.
+ */
+TEST(pose, transformRelativePoseToGlobal)
+{
+  const geometry_msgs::msg::Pose pose_relative = makePose(makePoint(4.9969, -28.1026, 0.214));
+  const geometry_msgs::msg::Pose pose_global = makePose(makePoint(81601.0571, 50099.0981, 34.595));
+  const geometry_msgs::msg::Pose pose_rel_global =
+    makePose(makePoint(81606.054, 50070.9955, 34.809));
+
+  EXPECT_POSE_EQ(
+    traffic_simulator::pose::transformRelativePoseToGlobal(pose_global, pose_relative),
+    pose_rel_global);
+}
+
+/**
+ * @note Test calculation correctness when only one pose is zeroed, the goal is to obtain a pose equal to the second argument.
+ */
+TEST(pose, relativePose_poses_zeros)
+{
+  const auto from = makePose(makePoint(0.0, 0.0, 0.0));
+  const auto to = makePose(makePoint(10.0, 51.0, 2.0));
+  const auto relative = traffic_simulator::pose::relativePose(from, to);
+
+  ASSERT_TRUE(relative.has_value());
+  EXPECT_POSE_EQ(relative.value(), to);
+}
+
+/**
+ * @note Test calculation correctness when both poses are zeroed.
+ */
+TEST(pose, relativePose_poses_zero)
+{
+  const auto from = makePose(makePoint(0.0, 0.0, 0.0));
+  const auto to = makePose(makePoint(0.0, 0.0, 0.0));
+  const auto relative = traffic_simulator::pose::relativePose(from, to);
+
+  ASSERT_TRUE(relative.has_value());
+  EXPECT_POSE_EQ(relative.value(), to);
+}
+
+/**
+ * @note Test calculation correctness with a non trivial example.
+ */
+TEST(pose, relativePose_poses_complex)
+{
+  const auto pose_relative = makePose(makePoint(4.9969, -28.1026, 0.214));
+  const auto from = makePose(makePoint(81601.0571, 50099.0981, 34.595));
+  const auto to = makePose(makePoint(81606.054, 50070.9955, 34.809));
+
+  const auto relative = traffic_simulator::pose::relativePose(from, to);
+
+  ASSERT_TRUE(relative.has_value());
+  EXPECT_POSE_NEAR(pose_relative, relative.value(), 0.01);
+}
+
+/**
+ * @note Test calculation correctness with the overload.
+ */
+TEST_F(PoseTest, relativePose_first)
+{
+  const auto pose_relative = makePose(
+    makePoint(9.9999, -0.0009, 0.0126),
+    geometry_msgs::build<geometry_msgs::msg::Quaternion>().x(0).y(0).z(0).w(1));
+  const auto from = makePose(
+    makePoint(81585.1622, 50176.9202, 34.2595),
+    geometry_msgs::build<geometry_msgs::msg::Quaternion>().x(0).y(0).z(-0.6397).w(0.7686));
+  const auto to =
+    traffic_simulator::helper::constructCanonicalizedLaneletPose(195, 10.0, 0.0, hdmap_utils);
+
+  const auto relative = traffic_simulator::pose::relativePose(from, to);
+
+  ASSERT_TRUE(relative.has_value());
+  EXPECT_POSE_NEAR(pose_relative, relative.value(), 0.01);
+}
+
+/**
+ * @note Test calculation correctness with the overload.
+ */
+TEST_F(PoseTest, relativePose_second)
```

Comment:
I believe "first" and "second" represent the order in which the allied functions were defined, but this could potentially change, making it inappropriate. Please change the names to reflect the intent or conditions of the test cases.

Reply:
Next time, please also fix pull request description.